airflow/providers/hashicorp/hooks/vault.py

```diff
@@ -0,0 +1,546 @@
+    """
+    Retrieves Authenticated client from Hashicorp Vault. This is purely internal class promoting
+    authentication code reuse between the Hook an Secret, it should not be used directly in
+    Airflow DAGs. Use VaultBackend for backend integration and Hook in case you want to communicate
+    with VaultHook using standard Airflow Connection definition.
+
+    :param url: Base URL for the Vault instance being addressed.
+    :type url: str
+    :param auth_type: Authentication Type for Vault. Default is ``token``. Available values are in
+        :py:const:`airflow.providers.hashicorp.hooks.vault.VALID_AUTH_TYPES`.
+    :type auth_type: str
+    :param mount_point: The "path" the secret engine was mounted on. Default is "secret". Note that
+         this mount_point is not used for authentication if authentication is done via a
+         different engine.
+    :type mount_point: str
+    :param kv_engine_version: Selects the version of the engine to run (``1`` or ``2``, default: ``2``).
+    :type kv_engine_version: int
+    :param token: Authentication token to include in requests sent to Vault
+        (for ``token`` and ``github`` auth_type).
+    :type token: str
+    :param username: Username for Authentication (for ``ldap`` and ``userpass`` auth_types).
+    :type username: str
+    :param password: Password for Authentication (for ``ldap`` and ``userpass`` auth_types).
+    :type password: str
+    :param secret_id: Secret ID for Authentication (for ``approle`` auth_type).
+    :type secret_id: str
+    :param role_id: Role ID for Authentication (for ``approle`` auth_type).
+    :type role_id: str
+    :param kubernetes_role: Role for Authentication (for ``kubernetes`` auth_type).
+    :type kubernetes_role: str
+    :param kubernetes_jwt_path: Path for kubernetes jwt token (for ``kubernetes`` auth_type, default:
+        ``/var/run/secrets/kubernetes.io/serviceaccount/token``).
+    :type kubernetes_jwt_path: str
+    :param gcp_key_path: Path to GCP Credential JSON file (for ``gcp`` auth_type).
+    :type gcp_key_path: str
+    :param gcp_scopes: Comma-separated string containing GCP scopes (for ``gcp`` auth_type).
+    :type gcp_scopes: str
+    """
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        url: Optional[str] = None,
+        auth_type: str = 'token',
+        mount_point: str = "secret",
+        kv_engine_version: Optional[int] = None,
+        token: Optional[str] = None,
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        secret_id: Optional[str] = None,
+        role_id: Optional[str] = None,
+        kubernetes_role: Optional[str] = None,
+        kubernetes_jwt_path: Optional[str] = '/var/run/secrets/kubernetes.io/serviceaccount/token',
+        gcp_key_path: Optional[str] = None,
+        gcp_scopes: Optional[str] = None,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
```

Comment:
```suggestion
        super().__init__()
```

Reply:
We don't want to pass kwargs to LogginMixin

Reply:
Good catch!